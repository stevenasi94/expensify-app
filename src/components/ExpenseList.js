import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpensesListItem";
import selectedExpense from "../selectors/expenses";
import ExpensesListFilters from "./ExpenseListFilters";

export const ExpenseList = (props) => (
    <div>
    {
        props.expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (
            props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />;
            })
        )
    }
</div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectedExpense(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);