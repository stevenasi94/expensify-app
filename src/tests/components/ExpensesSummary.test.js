import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummery } from "../../components/ExpenseSummary";

test("should correctly render ExpenseSummery with 1 expense", () => {
    const wrapper = shallow(<ExpenseSummery expenseCount={1} expenseTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpenseSummery with multiple expenses", () => {
    const wrapper = shallow(<ExpenseSummery expenseCount={23} expenseTotal={235123408987}/>);
    expect(wrapper).toMatchSnapshot();
});

