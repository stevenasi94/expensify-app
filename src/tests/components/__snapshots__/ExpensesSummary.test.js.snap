// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummery with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummery with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totaling 
    $2,351,234,089.87
  </h1>
</div>
`;
